src/Concerns/CapturesState.php

```diff
@@ -337,6 +338,22 @@ public function cacheEvent(CacheEvent $event): void
 
         [$record, $resolver] = $cacheEvent;
 
+        $rejectKeys = $this->captureDefaultVendorCacheKeys
+            ? $this->rejectCacheKeys
+            : [...$this->defaultVendorCacheKeys(), ...$this->rejectCacheKeys];
+
+        foreach ($rejectKeys as $reject) {
+            $isRegex = @preg_match($reject, '') !== false;
+
+            if ($isRegex && preg_match($reject, $record->key)) {
+                return;
+            }
+
+            if (! $isRegex && $record->key === $reject) {
+                return;
+            }
```

Comment:
I think this is effectively the same, but with one fewer `preg_match` call?

```suggestion
            $match = @preg_match($reject, $record->key);

            if ($match === 1) {
                return;
            }

            if ($match === false && $record->key === $reject) {
                return;
            }
```

Reply:
🧠🧠🧠
Updated in 79ecda9ba58173e46c5a6ea6b8e486c0d1a704e0